packages/plugin-core/src/commands/CreateNoteWithTraitCommand.ts

```diff
@@ -39,16 +40,38 @@ export class CreateNoteWithTraitCommand extends BaseCommand<
   CommandInput
 > {
   key: string;
-  trait: NoteTrait;
+  private _trait: NoteTrait | undefined;
+  private initTrait: () => NoteTrait;
   protected _extension: IDendronExtension;
 
-  constructor(ext: IDendronExtension, commandId: string, trait: NoteTrait) {
+  constructor(
+    ext: IDendronExtension,
+    commandId: string,
+    trait: NoteTrait | (() => NoteTrait)
```

Comment:
I'm not a big fan of having this OR'd type - I think what'd be cleaner is to remove this trait prop from the constructor, and then expose a `protected set trait(trait: NoteTrait)` method that each derived class like `MeetingNoteCommand` or `JournalNoteCommand` call as part of their run/execute.

If that's more than you'd like to refactor right now, can you add a TODO comment with this info?

Reply:
so the reason for passing in a function to the constructor is so we can register all commands at startup (vs waiting for the workspace to be available to register a command). my preference is for `trait` to be changed to `initTraitCb`. have it as an or type for now to avoid breaking tests and other interfaces. added a TODO statement to come back to this

Reply:
I understand the reasoning - my suggestion to move the setting of the trait to a setter and out of the constructor means that we also don't need to have the WS at init time, but only when the user runs the command.

Reply:
I'm not quite sure how that would be implemented. Regardless, I left a TODO here to change it - can we merge this and then work out details of the note trait command?